dsl/camel-jbang/camel-jbang-core/src/main/java/org/apache/camel/dsl/jbang/core/commands/infra/InfraRun.java

```diff
@@ -150,14 +137,43 @@ private void run(String testService, String testServiceImplementation) throws Ex
 
         printer().println(jsonMapper.writeValueAsString(properties));
 
+        String name = getLogFileName(testService, RuntimeUtil.getPid());
+        File logFile = new File(CommandLineHelper.getCamelDir(), name);
+        logFile.createNewFile();
+        logFile.deleteOnExit();
+
+        if (actualService instanceof ContainerService<?> cs) {
+            cs.followLog(new CamelLogConsumer(logFile.toPath(), logToStdout));
+        }
+
         if (!jsonOutput) {
-            printer().println("To stop the execution press q");
+            printer().println("Press any key to stop the execution");
         }
-        Scanner sc = new Scanner(System.in).useDelimiter("\n");
+        Scanner sc = new Scanner(System.in);
 
-        if (sc.nextLine().equals("q")) {
-            actualService.getClass().getMethod("shutdown").invoke(actualService);
-            sc.close();
+        while (!sc.hasNext()) {
         }
+
+        ((InfrastructureService) actualService).shutdown();
+        sc.close();
+    }
+
+    private static @NotNull DependencyDownloaderClassLoader getDependencyDownloaderClassLoader(
```

Comment:
Just a heads up that there is a `@NotNull` annotation here (IntelliJ keeps adding those from time to time).

Reply:
thanks, I removed that